StyleCop.Analyzers/StyleCopTester/Program.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace StyleCopTester
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.MSBuild;
+
+    /// <summary>
+    /// StyleCopTester is a tool that will analyze a solution, find diagnostics in it and will print out the number of
+    /// diagnostics it could find. This is useful to easily test performance without having the overhead of visual
+    /// studio running.
+    /// </summary>
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            // A valid call must have at least two parameters. A solution file and /all or one /id:SAXXXX
+            if (args.Length <= 1)
+            {
+                PrintHelp();
+            }
+            else
+            {
+                Stopwatch stopwatch = Stopwatch.StartNew();
+                var analyzers = GetAllAnalyzers();
+
+                analyzers = FilterAnalyzers(analyzers, args).ToImmutableArray();
+
+                if (analyzers.Length == 0)
+                {
+                    PrintHelp();
+                    return;
+                }
+
+                MSBuildWorkspace workspace = MSBuildWorkspace.Create();
+                Solution solution = workspace.OpenSolutionAsync(args[0]).Result;
+
+                Console.WriteLine($"Loaded solution in {stopwatch.ElapsedMilliseconds}ms");
+
+                stopwatch.Restart();
+
+                var diagnostics = GetAnalyzerDiagnosticsAsync(solution, args[0], analyzers).Result;
+
+                Console.WriteLine($"Found {diagnostics.Count} diagnostics in {stopwatch.ElapsedMilliseconds}ms");
+            }
+        }
+
+        private static IEnumerable<DiagnosticAnalyzer> FilterAnalyzers(IEnumerable<DiagnosticAnalyzer> analyzers, string[] args)
+        {
+            bool useAll = args.Contains("/all");
+
+            HashSet<string> ids = new HashSet<string>(args.Where(y => y.StartsWith("/id:")).Select(y => y.Substring(4)));
```

Comment:
More obviously skip specified diagnostics if both `/all` and `/id` options presented.

    StyleCopTester /all /id:SA1010 /id:SA1011